### ISSUE TITLE
Rides with "Synchronize with adjacent stations" stall out

### ISSUE THREAD

**Issue by Superjustinbros** [OpenRCT2/OpenRCT2#10655]
**OS:** [e.g. Windows 7]
**Version:** 0.2.4-485-g3f837f561
**Commit/Build:** 3f837f5

In vanilla RCT and RCT2, I recall that if you were to build a track design that has two stations adjacent to and synchronized with each other on the same coaster, all the trains would only spawn on one station and to remedy this, the coaster would send one or several trains over so both stations have an even number of (or at least one) trains.

In my last few play sessions, I've had situations where a ride never operated because there was never a second train in the adjacent station to accompany the first, requiring me to de-check "Synchronize with adjacent stations" before re-checking it later.

**Steps to reproduce:**
1. Create a "dueling roller coaster" or place a track design of one down
2. Make sure Synchronize with Adjacent Stations is checked
3. Test or open the coaster, notice that regardless of what options are selected for when the train should depart, the trains in the first station never budge unless synchronization is turned off.

**Comment by ocalhoun6** [comment]
I'm about 90% certain that RCT1/2 didn't do this automatically either -- you had to turn synchronize stations off long enough to launch a train or to, then turn it back on manually.

_But_ this might be a good feature request to add to OpenRCT2.

It might be a little tricky, but I think it should in theory be possible to have the synchronization code check to see if all the synchronized stations are part of the same ride and, if so, allow the first (number of trains / number of stations) to leave the station without waiting for a synchronous departure, continuing until all stations have an equal number of trains parked.

Then there's the special case where two or more synchronized stations are part of a mobius coaster, but a third (or more) stations in the same synchronous state are _not_ part of that coaster... But dealing with a situation as complicated as that seems a bit too much to ask.

(I'll also note that under certain conditions, trains can get stuck this way after a ride breakdown, so it's not _only_ an issue when first constructing the ride.)

**Comment by awebeer256** [comment]
I was thinking about this today (I found this issue by searching, otherwise I would've made a new one) and it occurred to me that I would prefer rides to _always_ spawn trains evenly distributed between stations. As far as I can tell, there are three use cases for a ride having multiple stations: 1) it's a mobius racing ride, 2) it's (functionally) a transport ride, 3) guest loading and unloading are handled at different stations (e.g. the Steeplechase pre-built design Pony Tales).

In case 1, this change would be a huge improvement, as was discussed in previous comments.

In case 2, it would be a small improvement. In a transport ride, you generally want all the trains to be evenly distributed around the track during normal operation, and distributing them among the stations is a much closer approximation to this than starting all of them in Station 1 is. The reason it's only a **small** improvement is that properly set minimum and maximum waiting times will spread the trains out regardless.

In case 3, this change would almost always make basically no difference, since the trains would just move up to the loading station immediately anyway.

And while we're on the subject, I think the game should determine a ride's maximum number of trains based on how many can fit in _all stations combined,_ rather than by how many can fit in one station (obviously, this doesn't apply in block-sectioned operating modes, nor to rides that determine the maximum number of trains by the length of the track).

**Comment by ocalhoun6** [comment]
>And while we're on the subject, I think the game should determine a ride's maximum number of trains based on how many can fit in all stations combined, rather than by how many can fit in one station

This could backfire pretty hard, though.

Imagine you have a coaster acting as a transport ride. It has 3 stations that can hold 2 trains each, for a total of 6. And maybe it's set to leave as soon as the next train arrives at the station. A new train arrives, triggering the first train to leave ... but then a 'restraints stuck closed' breakdown occurs. The first train _doesn't_ leave the station, and now the station is completely full. And suppose that there's already a train halfway through the track approaching this same station, and they enter at a high speed. When the third train arrives, there will be no room for it and it will slam into the previous train at full speed, causing a crash and lots of dead peeps. Hell, if the stations are poorly spaced, it's possible that all 6 trains will end up stopping at one station.

**Comment by pizza2004** [comment]
Given that there are no possibilities for things like extra trains that only operate at specific times, like such a ride might have in real life, I think that exact scenario is just something that the player should have to consider.  Alternatively, adding such extra stations for transport rides could be an interesting addition.

**Comment by awebeer256** [comment]
> I think that exact scenario is just something that the player should have to consider.

I agree with this. If you can't be arsed to just use block-sectioned mode (a.k.a the proper solution to this problem :P), then put some brakes and a chain lift track piece before each station.

> adding such extra trains in a side station for transport rides could be an interesting addition. Perhaps they would only trigger during a special event and they could just merge on and off the track.

That's, uh, quite a bit of mission creep you've got there.
